storage/replica_command.go

```diff
@@ -546,12 +550,29 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 		// less than "f", and so the meta row {"f"->["c", "f")} would be
 		// ignored by MVCCReverseScan. In the second case, the range descriptor
 		// received by MVCCScan will be filtered before results are returned.
+		// Example: ranges ["c", "f") and ["f", "z"), reverse scan on ["d", "f"]
+		// leads to inspecting the descriptor {"z"->["f","z")}, which is
+		// discarded since it's not being asked for.
 		startKey, endKey := keys.MetaScanBounds(args.Key)
 		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, 1,
 			args.Timestamp, consistent, args.Txn)
 		if err != nil {
 			return reply, nil, err
 		}
+		var firstRD proto.RangeDescriptor
+		if err = gogoproto.Unmarshal(kvs[0].Value.Bytes, &firstRD); err != nil {
+			return reply, nil, err
+		}
+		if keys.RangeMetaKey(firstRD.StartKey).Less(args.Key) {
```

Comment:
Hmm, I thought that we want to return a range that contains `b` (and following ranges if `args.MaxRanges` is > 1) when `Args.Key` is `\meta2b`. Here is the comment in the `RangeLookupRequest` definition:

```
// A RangeLookupRequest is arguments to the RangeLookup() method. It
// specifies the key for which the containing range is being
// requested, and the maximum number of range descriptors that should
// be returned, if there are additional consecutive addressable
// ranges.
```

For example, `rangeGCQueue.process()` sets `Args.Key` to the meta key of the range start key (not a reverse lookup, though).